### PR TITLE
Set Obsoletes: for distro-provided package names

### PR DESCRIPTION
**Pull request by mjmac**
May improve dependency resolution.

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -39,5 +39,6 @@
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
 //@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@bmurrell/package-building-test-daos-matrix")
 
 packageBuildingPipelineDAOS(['distros': ['centos7', 'el8', 'leap15']])
```
Comment (brianjmurrell) [inline Jenkinsfile]:
```suggestion
packageBuildingPipelineDAOS(['distros': ['centos7', 'el8', 'leap15'],
                             'rpmlint_rpms_allow_errors': true])
```

**Comment by brianjmurrell** [comment]
I don't get it.  I don't see how [this result](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fgo-build-toolchain/detail/PR-5/16/pipeline/) is any different than [this result](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fmercury/detail/PR-77/56/pipeline/).  They both have stages which have `rpmlint`ing errors converted to warnings by `                             'rpmlint_rpms_allow_errors': true`.

Yet [the other] one allows the other stages to be `SUCCESS`ful and this one makes one of the stages `UNSTABLE` as a result of a different stage being `UNSTABLE`.

**Comment by brianjmurrell** [comment]
@mjmac This passes with pipeline-lib@master now.  Handing it back over to you.

**Comment by mjmac** [comment]
Not necessary. This was an experiment to see if it would solve the problem ultimately solved by disabling the go-toolkit module in dnf.

**Pull request closed without merging**